streams/examples/main/java/org/apache/kafka/streams/examples/WordCount/WordCountProcessorJob.java

```diff
@@ -49,17 +48,21 @@
                 public void init(ProcessorContext context) {
                     this.context = context;
                     this.context.schedule(1000);
-                    this.kvStore = (KeyValueStore<String, Integer>) context.getStateStore("LOCAL-STATE");
+                    this.kvStore = (KeyValueStore<String, Integer>) context.getStateStore("Counts");
                 }
 
                 @Override
-                public void process(String key, String value) {
-                    Integer oldValue = this.kvStore.get(key);
-                    Integer newValue = Integer.parseInt(value);
-                    if (oldValue == null) {
-                        this.kvStore.put(key, newValue);
-                    } else {
-                        this.kvStore.put(key, oldValue + newValue);
+                public void process(String dummy, String line) {
+                    String words[] = line.toLowerCase().split(" ");
+
+                    for (String word : words) {
+                        Integer oldValue = this.kvStore.get(word);
+
+                        if (oldValue == null) {
```

Comment:
Not related to this patch, but would be nice if `KeyValueStore` had `computeIfAbsent`. Is this just missing because Java 7 collections don't provide this? I'd imagine this is an extremely common pattern...